### PR TITLE
Pushing Changes from our private repo to the public repo

### PR DESCRIPTION
**Pull request by J314Y**
@pdeshikachar @dialtone @walterking

### CODE REVIEW

**Comment by yixu34** [comment]
Er this has some stuff in reference.yml that shouldn't be there.

File: reference.yml [hunk 1]
```diff
@@ -19,13 +19,14 @@ server:
...
 shiro:
-  iniConfigs: ["classpath:shiro_allow_all.ini"]
+  iniConfigs: ["classpath:shiro_adroll.ini"]
 
 dataSourceFactory:
   driverClass: com.mysql.jdbc.Driver
-  user: MYSQL_USER
-  password: MYSQl_PASSWORD
-  url: jdbc:mysql://127.0.0.1:3306/MYSQL_DB_NAME
+  user: hive
```
Comment (yixu34) [inline reference.yml]: These three lines should be redacted.

File: src/main/java/com/airbnb/airpal/api/output/persistors/S3FilePersistor.java [hunk 2]
```diff
@@ -35,7 +37,7 @@ public boolean canPersist(QueryExecutionAuthorizer authorizer)
 
     private String getOutputKey(String fileBaseName)
     {
-        return "airpal/" + fileBaseName;
+        return "presto/airpal/tmp/" + fileBaseName;
```
Comment (yixu34) [inline src/main/java/com/airbnb/airpal/api/output/persistors/S3FilePersistor.java]: This too.

File: src/main/java/com/airbnb/airpal/modules/AirpalModule.java [hunk 3]
```diff
@@ -257,23 +260,20 @@ public EventBus provideEventBus(@Named("event-bus") ExecutorService executor)
...
-        if (awsCredentials == null) {
-            return new AmazonS3Client();
-        }
+        // Altered so that awsCredentials is not used to construct the S3Client
+        // This utilizes the amazon api to control token refresh and their libraries
+        // to handle passing credentials around.
 
-        return new AmazonS3Client(awsCredentials);
+        AmazonS3Client s3c = new AmazonS3Client(new DefaultAWSCredentialsProviderChain());
+        s3c.setRegion(Region.getRegion(Regions.US_WEST_2));
```
Comment (yixu34) [inline src/main/java/com/airbnb/airpal/modules/AirpalModule.java]: And this.

File: src/main/java/com/airbnb/airpal/resources/ResultsPreviewResource.java [hunk 4]
```diff
@@ -65,7 +65,7 @@ public Response getFile(@QueryParam("fileURI") URI fileURI,
 
     private String getOutputKey(String fileBaseName)
     {
-        return "airpal/" + fileBaseName;
+        return "presto/airpal/tmp/" + fileBaseName;
```
Comment (yixu34) [inline src/main/java/com/airbnb/airpal/resources/ResultsPreviewResource.java]: Redact

File: src/main/java/com/airbnb/airpal/resources/S3FilesResource.java [hunk 5]
```diff
@@ -37,7 +37,7 @@ public S3FilesResource(
 
     private String getOutputKey(String fileBaseName)
     {
-        return "airpal/" + fileBaseName;
+        return "presto/airpal/tmp/" + fileBaseName;
```
Comment (yixu34) [inline src/main/java/com/airbnb/airpal/resources/S3FilesResource.java]: Redact

File: reference.yml [hunk 6]
```diff
@@ -19,13 +19,14 @@ server:
...
+  password: hive
+  url: jdbc:mysql://hive-metastore.c3fdiq8snpad.us-west-2.rds.amazonaws.com:3306/airpal
```
Comment (walterking) [inline reference.yml]: this is safe to be public?

Comment (J314Y) [inline reference.yml]: Nope

**Comment by yixu34** [comment]
Looks good now. We should delete all of the comments on this PR after everyone OK's it.

**Comment by J314Y** [comment]
@yixu34 @wesleychowadroll @walterking @pdeshikachar These are the latest changes. if it looks good to you I can manually merge since there are a few conflicts

**Comment by wesleychowadroll** [comment]
LGTM

**Pull request merged**